src/metal/scheduler.ts

```diff
@@ -28,26 +28,45 @@ const TOKEN_SEED = 'xxxx'.replace(/[xy]/g, function(c) {
   return v.toString(16);
 });
 let tokenCounter = 0;
-
-function generateRandomToken() {
-  return Math.floor(Math.random() * (9999999 - 0o0)).toString(16);
-}
+let elementSchedulerCounter = 0;
 
 export class Frame implements FrameInterface {
   constructor(
     public timestamp: number,
     public scrollTop: number,
     public scrollLeft: number,
     public width: number,
-    public height: number
+    public height: number,
+    public x: number,
+    public y: number,
+    public top: number,
+    public left: number
   ) {}
-  static generate(): Frame {
+  static generate(root?: Element): Frame {
+    if (root && document.documentElement.contains(root)) {
```

Comment:
This guard, and the guard contained with `getBoundingClientRect` may be redundant when used together. As if the `getBoundingClientRect` fails, the various unknown attributes become `0,0,0,0` already right?